src/benchmarklib/benchmark_utils.cpp

```diff
@@ -364,16 +364,17 @@ nlohmann::json EncodingConfig::to_json() const {
 }
 
 void BenchmarkTableEncoder::encode(const std::string& table_name, const std::shared_ptr<Table>& table,
-                                   const EncodingConfig& config) {
-  const auto& type_mapping = config.type_encoding_mapping;
-  const auto& custom_mapping = config.custom_encoding_mapping;
+                                   const EncodingConfig& encoding_config, std::ostream& out) {
```

Comment:
Mmh, should an encoder really print things? How about return a bool and handling the "failure" closer to the caller?

Reply:
I know where you come from, but this is not a "failure" it's just something the user of the benchmark might want to know/should be informed about.

Reply:
Also valid. In that case, do we ever expect to use the `out` parameter? As it is defaulted to `cout` anyway, I don't see the advantage of passing it around here.

Reply:
I don't like this function logging stuff either, but I see no practical alternative

Reply:
It will be the benchmark log file, if the user specified one